sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -796,3 +797,64 @@ case class SchemaOfJson(
 
   override def prettyName: String = "schema_of_json"
 }
+
+/**
+ * A function which returns all the keys of outer JSON object.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(json_object) - returns all the keys of outer JSON object.",
+  arguments = """
+    Arguments:
+      * json_object - A JSON object. If it is an invalid string, the function returns null.
+          If it is a JSON array or null, a runtime exception will be thrown.
+  """,
+  examples = """
+    Examples:
+      > Select _FUNC_('{}');
+        []
+      > Select _FUNC_('{"key": "value"}');
+        ["key"]
+      > Select _FUNC_('{"f1":"abc","f2":{"f3":"a", "f4":"b"}}');
+        ["f1","f2"]
+  """,
+  since = "3.1.0")
+case class JsonObjectKeys(child: Expression) extends UnaryExpression with CodegenFallback {
+  override def dataType: DataType = ArrayType(StringType)
+  override def nullable: Boolean = true
+  override def prettyName: String = "json_object_keys"
+
+  override def eval(input: InternalRow): Any = {
+    try {
+      val json = child.eval(input).asInstanceOf[UTF8String]
+      Utils.tryWithResource(CreateJacksonParser.utf8String(SharedFactory.jsonFactory, json)) {
+        parser => getJsonKeys(parser, input)
+      }
+    } catch {
+      case _: JsonProcessingException => null
+    }
```

Comment:
Apparently, the current implementation seems to have the same limitations: `NullPointerException` and `ClassCastException`. Please add more unit test cases with `null` and `Integer` input.

Reply:
I checked.
```
scala> sql("select json_object_keys(null)").show
java.lang.NullPointerException

scala> sql("select json_object_keys(1)").show
java.lang.ClassCastException: class java.lang.Integer cannot be cast to class org.apache.spark.unsafe.types.UTF8String (java.lang.Integer is in module java.base of loader 'bootstrap'; org.apache.spark.unsafe.types.UTF8String is in unnamed module of loader 'app')
```